Unity-C#/README.md

```diff
@@ -0,0 +1,263 @@
+```
+
+
+#### Events
+```csharp
+public ChangedNameHandler OnNameChanged
+```
+
+
+#### Specific Naming Conventions
+
+- The "is" prefix SHOULD be used for boolean variables and methods. Alternatives include "has", "can" and "should".
+
+- The terms "initialize" or "init" CAN be used where an object or a concept is established.
+
+- Plural form MUST be used to name collections.
+
+- Iterator variables SHOULD be called "i", "j", "k", etc.
+
+- Complement names MUST be used for complement entities. Examples: get/set, add/remove, create/destroy, start/stop, insert/delete, begin/end, etc.
+
+- Negated boolean variable names MUST be avoided:
+```csharp
+	bool isNotError, isNotFound; // WRONG.
+```
+
+- Methods returning an object MAY be named after what they return, and methods returning void after what they do.
+
+
+## Variable declarations
+Declare one variable per type statement, it makes it easier to re-order the lines, and put declarations in the order in which they make sense.
+Ideally, variables should be initialized where they are declared and they must be declared in the smallest scope possible.
+
+
+### Loop
+Only loop control statements MUST be included in the "for" loop construction.
+
+Loop variables should be initialized immediately before the loop; loop variables in a "for" statement may be initialized in the "for" loop construction.
+
+The use of "break" and "continue" is not discouraged.
+
+### Conditionals
+Complex conditional expressions SHOULD be avoided; use temporary boolean variables instead with proper naming.
+
+```csharp
+bool isAuthorized = (user.IsAdmin() || user.IsModerator());
+if (isAuthorized)
+{
+	Console.Log('winning');
+}
+```
+
+The nominal case should be put in the "if" part and the exception in the "else" part of an "if" statement.
+
+Executable statements in conditionals MUST be avoided.
+
+### Miscellaneous
+The use of magic numbers in the code should be avoided; they should be declared using named **constants** instead.
+
+
+## Return statement
+To avoid deep nesting of if-statements, always return a function's value as early as possible.
+
+```csharp
+public bool IsPercentage(int percent) 
+{
+	if (percent < 0)
+	{
+		return false;
+	}
+
+	if (percent > 100)
+	{
+		return false;
+	}
+
+	return true;
+}
+
+```
+
+
+## Comments
+
+Comments are the documentation of your code; therefore **tricky code should not be commented but rewritten**. 
+
+All comments should be written in English and be politically correct, even those that are supposed to be temporary.
+
+Comments should be indented relative to their position in the code, preceding the code in question.
+
+Even if everybody likes commented code, do not comment every single line of code; you will kill the readability of it.
+
+Comments are supposed to explain an algorithm, not repeat it in a different language. This is wrong:
+```csharp
+// increment i by 1
+i += 1;
+```
+
+Try to put comment at the top of a block of code to explain the point of it..
```

Comment:
I absolutely have to agree with you on most points you make here, @daniel-dressler . Comments are not supposed to rewrite the code in human-readable-language. We're programmers. Code is already supposed to be human readable. That's the whole point of a style guide: ensure that the code is self-explanatory. The utopian code base indeed carries no comments (but I don't think we'll live to see this happen). Variable names, argument names, function names, everything should describe its own reason to exist and what it does through the virtue of a well chosen name.

So when should we comment? When the language won't let you do all you need to communicate some of what I wrote above (when strong typing is lacking for example, because a type does communicate). That's why we use JSDoc in JavaScript. Comments are also useful when you're about to write a function that needs to do something complicated. You don't repeat the algorithm (it will get out of sync with reality). You describe what it does (not how it does it, the code *should* suffice). Also, I believe one should *never* repeat values in comments (eg: `int length = 5; // length is 5`), because the moment they're out of sync, you have nothing to fall back to in order to discover which is true (is this a bug, or just a bad comment?).

There are *always* going to be exceptions to these "rules". Performance sensitive code sometimes just ends up being ugly. I'm sure OS kernels and game engines are full of it. That's when you comment the crap out of the solution to make it really, really, really clear what that ugly piece of code achieves and how it does that. And you apologize to the next maintainer :)

TL;DR: Ideally, comments describe what and code describes how. They should not tell the same story twice.